src/kernels/MIOpenSoftmaxAttn.cpp

```diff
@@ -408,15 +409,15 @@ extern "C" __global__ void ScaleReduce(const float* __restrict__ in,
 
     auto in_ptr    = in + gid;
     auto out_ptr   = out + gid;
-    const auto end = in_ptr + nhsd;
+    const auto end = in + nhsd;
```

Comment:
This seems like a significant change? How does in relate to in_ptr? Was this a bug before?

Reply:
It was a bug. 'end' pointer counts from the beginning - from `begin` pointer (which is `in`).
`in_ptr` is a shifted by global thread id, It can index current element, but can't be used for the `end` calculatons.